xcube/webapi/context.py

```diff
@@ -456,6 +456,14 @@ def get_dataset_configs_from_stores(self, data_store_pool: DataStorePool) \
                         StoreInstanceId=store_instance_id,
                         **dataset_config_base
                     )
+                    if dataset_config.get('Identifier') is not None:
+                        if dataset_config['Path'] == store_dataset_id:
+                            # we will use the preconfigured identifier
+                            all_dataset_configs.append(dataset_config)
+                            continue
+                        LOG.info('Pre-configured identifier is ignored as '
+                                 'the configured path does not unambiguously '
+                                 'denote a dataset')
```

Comment:
Allow setting user-defined identifiers only if the path does not contain wildcard characters '*' or '?'. If it does, raise a configuration error, because such a configuration is invalid ("User-defined identifiers can only be assigned to datasets with non-wildcard paths.").